lib/commands/app-management.js

```diff
@@ -81,8 +81,6 @@ commands.activateApp = async function activateApp (appId) {
     'am', (apiLevel < 26) ? 'start' : 'start-activity',
     '-a', 'android.intent.action.MAIN',
     '-c', 'android.intent.category.LAUNCHER',
-    // FLAG_ACTIVITY_REORDER_TO_FRONT | FLAG_ACTIVITY_BROUGHT_TO_FRONT | FLAG_ACTIVITY_NEW_TASK
```

Comment:
Or probably `-f 0x01000000`? https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_PREVIOUS_IS_TOP

I haven't read the Android code base yet. (Will do later)

Reply:
With appium's exmample app. The below kept open activity to bring it to the forwarground.

```
% adb shell am start-activity -W -n io.appium.android.apis/.ApiDemos  -a android.intent.action.MAIN -c android.intent.category.LAUNCHER
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=io.appium.android.apis/.ApiDemos }
Warning: Activity not started, its current task has been brought to the front
Status: ok
LaunchState: HOT
Activity: io.appium.android.apis/.view.WebView1
TotalTime: 122
WaitTime: 129
Complete
% adb shell am start-activity -W -n io.appium.android.apis/.ApiDemos  -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -f 0x01000000
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x1000000 cmp=io.appium.android.apis/.ApiDemos }
Warning: Activity not started, its current task has been brought to the front
Status: ok
LaunchState: HOT
Activity: io.appium.android.apis/.view.WebView1
TotalTime: 132
WaitTime: 138
Complete
```

but at least `0x10420000` reset the activity as these flags' meaning.
I mean `adb shell am start-activity -W -n io.appium.android.apis/.ApiDemos  -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -f 0x10420000` did not bring `io.appium.android.apis/.view.WebView1` as the top even the output was the below.

```
% adb shell am start-activity -W -n io.appium.android.apis/.ApiDemos  -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -f 0x10420000
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10420000 cmp=io.appium.android.apis/.ApiDemos }
Warning: Activity not started, intent has been delivered to currently running top-most instance.
Status: ok
LaunchState: UNKNOWN (0)
Activity: io.appium.android.apis/.view.WebView1
TotalTime: 0
WaitTime: 13
Complete
```

Reply:
This `--activity-single-top` instead of `-n` worked as same.

`adb shell am start-activity --activity-single-top  io.appium.android.apis/.ApiDemos  -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -f 0x10420000`

Reply:
hm, we should keep the monkey way, or use `am start` way with some changes to keep resume behavior as possible, or follow the current master way as a breaking change. (we should address this behavior, activities will be killed by activateApp, in the readme, i think)

Will continue to consider tomorrow.

Reply:
mm, https://android.googlesource.com/platform/development/+/ecc1a3d0cf3af3dbbf4c190e1060fc1c0f8fb8a6/cmds/monkey/src/com/android/commands/monkey/MonkeyActivityEvent.java?autodive=0%2F#43 probably we can keep this flag for `am start` as well...?

Reply:
Maybe https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_NEW_TASK should work as resume ...? at least. Or `0x10200000` (with `FLAG_ACTIVITY_RESET_TASK_IF_NEEDED` as well).

In my testing, `0x10200000` and `0x10000000` kept current activity, but `0x10420000` reset it.

Reply:
let it be `0x10200000 ` to mimic the original task manager behavior